// -*- Protocol-Buffers -*-
// ==================================================================================================
//
//
// =================================================================================================
// @file hardwareControl.proto
// @brief Service definitions for gRPC
// @author Andrew Kessler <andrewkessler6@gmail.com>

syntax = "proto3";
import "google/protobuf/timestamp.proto";


package hardwarecontrol;

service HardwareControl {

    rpc Echo (EchoRequest) returns (EchoResponse) {}
    rpc SetRelayState (RelayState) returns (Empty) {}
    rpc GetRelayStates (Empty) returns (RelayStates) {}

    rpc GetTemperature (Empty) returns (Temperature) {}

}

message Empty {

}

message EchoRequest {
    string payload =1;
}

message EchoResponse {
    string payback = 1;
}

message RelayState {
    uint32 channel = 1;
    bool isEngaged = 2;
}

message RelayStates {
    repeated RelayState states = 1;
}

message Temperature {
    float temperature_degC = 1;
}